File: parsingScripts/evenDelimiters.py
import os

directory = '../contents'

def count_delimiters(line):
    count = 0
    ignore_next = False
    for char in line:
        if char == '\\':
            ignore_next = True
        elif char == '$' and not ignore_next:
            count += 1
        else:
            ignore_next = False
    return count

def process_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        for line_number, line in enumerate(file, 1):
            delimiter_count = count_delimiters(line)
            if delimiter_count % 2 != 0:
                print(f"Flagged line in {file_path} at line {line_number}: {line}")

def parse_directory(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            file_path = os.path.join(root, file)
            process_file(file_path)

parse_directory(directory)

File: parsingScripts/evenDelimiters.js
import fs from 'fs';
import path from 'path';

const directory = 'contents';

function countDelimiters(line) {
    let count = 0;
    let ignoreNext = false;
    for (const char of line) {
        if (char === '\\') {
            ignoreNext = true;
        } else if (char === '$' && !ignoreNext) {
            count += 1;
        } else {
            ignoreNext = false;
        }
    }
    return count;
}

function processFile(filePath) {
    fs.readFile(filePath, 'utf-8', (err, data) => {
        if (err) {
            console.error(err);
            return;
        }

        const lines = data.split('\n');
        for (let lineIndex = 0; lineIndex < lines.length; lineIndex++) {
            const line = lines[lineIndex];
            const lineNumber = lineIndex + 1;
            const delimiterCount = countDelimiters(line);
            if (delimiterCount % 2 !== 0) {
                console.log(`Flagged line in ${filePath} at line ${lineNumber}: ${line}`);
            }
        }
    });
}

function parseDirectory(directoryPath) {
    fs.readdir(directoryPath, (err, files) => {
        if (err) {
            console.error(err);
            return;
        }

        files.forEach((file) => {
            const filePath = path.join(directoryPath, file);
            processFile(filePath);
        });
    });
}

parseDirectory(directory);
